packages/modules/vehicles/polestar/api.py

```diff
@@ -0,0 +1,106 @@
+        access_token = self.auth.get_auth_token()
+        if access_token is None:
+            log.error("query_params:not yet authenticated")
+            return None
+
+        headers = {
+            "Content-Type": "application/json",
+            "authorization": f"Bearer {self.auth.access_token}"
+        }
+
+        log.info("query_params:%s", params['query'])
+        try:
+            result = self.client_session.get(url=url, params=params, headers=headers)
+        except requests.RequestException as e:
+            log.error("query_params:http error:%s", e)
+            return None
+
+        if result.status_code == 401:
+            self.auth.delete_token()
+            log.error("query_params error: get response %d: unauthorized Exception", result.status_code)
+            return None
+        if result.status_code != 200:
+            log.error("query_params error: get response %d: %s", result.status_code, result.text)
+            return None
+
+        result_data = result.json()
+        if result_data.get('errors'):
+            error_message = result_data['errors'][0]['message']
+            log.error("query_params error: %s", error_message)
+            return None
+
+        log.debug(result_data)
+        return result_data
+
+    def get_battery_data(self) -> dict or None:
+        params = {
+            "query": "query GetBatteryData($vin: String!) { getBatteryData(vin: $vin) { "
+            + "batteryChargeLevelPercentage  estimatedDistanceToEmptyKm } }",
+            "operationName": "GetBatteryData",
+            "variables": "{\"vin\":\"" + self.vin + "\"}"
+        }
+
+        result = self.query_params(params)
+
+        if result is not None and result['data'] is not None and result['data']['getBatteryData'] is not None \
+                and result['data']['getBatteryData']['batteryChargeLevelPercentage'] is not None:
+            return result['data']['getBatteryData']
+        elif self.auth.access_token is not None:
+            # if we got an access code but the query failed, VIN could be wrong, so let`s check it
+            self.check_vin()
+            return None
+
+    def check_vin(self) -> None:
+        # get Vehicle Data
+        params = {
+            "query": "query GetConsumerCarsV2 { getConsumerCarsV2 { vin internalVehicleIdentifier __typename }}",
+            "operationName": "GetConsumerCarsV2",
+            "variables": "{}"
+        }
+        result = self.query_params(params, url='https://pc-api.polestar.com/eu-north-1/my-star/')
+        if result is not None and result['data'] is not None:
+            vins = []
+            # get list of cars and store the ones not matching our vin
+            cars = result['data']['getConsumerCarsV2']
+            if len(cars) == 0:
+                log.error("No vehicles found using specified account. Did you use the Polestar owners account?")
+                return None
```

Comment:
```suggestion
                raise Exception("No vehicles found using specified account. Did you use the Polestar owners account?")
```
Hier auch eine Exception werfen, dann kommt die Meldung auch im Frontend beim Benutzer an.
Bitte dann auch eine deutsche Meldung, da das Frontend auf Deutsch ist.

Reply:
Ok, ich bin mit dem Github Web Interface nicht so vertraut, wie kann ich die deutsche Übersetzung einpflegen?

Reply:
```suggestion
                raise Exception("Es konnten keine Fahrzeuge im Account gefunden werden. Bitte in den Einstellungen prüfen, ob der Besitzer-Account des Polestars eingetragen ist.")
```